node/libs/crypto/src/bn254/tests.rs

```diff
@@ -89,6 +89,18 @@ fn aggregate_signature_failure_smoke() {
     assert!(agg.verify(pks.iter().map(|pk| (&msg[..], pk))).is_err())
 }
 
+#[test]
+fn pk_validity_checks() {
+    // Check that we can't decode the null point.
+    let pk = PublicKey::default();
+    let bytes = pk.encode();
+    let pk_decoded = PublicKey::decode(&bytes);
+    let err = pk_decoded.unwrap_err();
+    assert_eq!(format!("{}", err.root_cause()), "Public key can't be zero");
+
+    // TODO: Create a G2 point in the wrong subgroup and test against it.
```

Comment:
I suppose you can sample at random public key material, until you find a point in the wrong subgroup and then pass it to decode() to check that it also returns an error.

Reply:
or sample and hardcode into the test.

Reply:
Nope. Random G2 generation always [creates points in the correct subgroup](https://github.com/matter-labs/pairing/blob/master/src/bn256/ec.rs#L1149).
There's a [test](https://github.com/matter-labs/pairing/blob/master/src/bn256/ec.rs#L1522) that I can port. I'll probably do it, just not today.

Reply:
not random g2, I mean random slice of bytes

Reply:
@brunoffranca reading the code you referenced, it always creates a point from a random `x` because a loop is used. As mentioned in https://github.com/matter-labs/era-consensus/pull/80, it's about 50% chance.